src/vm/comutilnative.cpp

```diff
@@ -2624,14 +2626,107 @@ FCIMPL6(INT32, ManagedLoggingHelper::GetRegistryLoggingValues, CLR_BOOL* bLoggin
 }
 FCIMPLEND
 
+static BOOL HasOverriddenEqualsWorker(MethodTable* mt)
+{
+    MethodTable::IntroducedMethodIterator it(mt);
+    for (; it.IsValid(); it.Next())
+    {
+        if (strcmp(it.GetMethodDesc()->GetName(), "Equals") == 0)
```

Comment:
Seems like we can just check is it from overriding via `IsVirtual()` here. Because struct is sealed, user has no chance to introduce a new virtual method. So if the method is virtual, it is from parent. 

Note I'm not considering the scenario that user changes the override method name from IL